backend/src/models/shared.rs

```diff
@@ -0,0 +1,111 @@
+use nutype::nutype;
+use serde::{Deserialize, Serialize};
+use std::{net::SocketAddr, time::Duration};
+use utoipa::{IntoParams, ToSchema};
+
+#[nutype(sanitize(trim, lowercase) validate(not_empty, regex = r"(https?)?(?::[\/]{2})?([^\:]+):(([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$)"))]
+#[derive(Display, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub struct Hostname(String);
```

Comment:
Probably, it should wrap `Uri`. In this way all the validations will done by that type, additionally it would be much safer and simpler to access the port number

Reply:
agree